Makefile

```diff
@@ -380,9 +380,14 @@ use-setup-envtest:
 
 ENVTEST_TIMEOUT ?= 5m
 
+# TODO: remove when Gateway API v1 gets released and updated in go.mod
+.PHONY: go-get-gateway-api
+go-get-gateway-api:
+	go mod download -x $(GATEWAY_API_PACKAGE)@v1.0.0-rc2
```

Comment:
This is needed because we install Gateway API CRDs from the Go module cache ( to avoid downloading those again since they are readily available on the system ) https://github.com/Kong/kubernetes-ingress-controller/blob/513db87cbf94ce66207f74365b502e8cde841357/test/envtest/setup.go#L130-L135.

This is an issue when we want to use a Go module tag (that's not a semver tag) like `v1.0.0-rc2.0.20231030133349-8f1b718543ca` with the `replace` in `go.mod`.

This will go away when we use v1 version and remove the `replace`.

The CRD difference between the rc and the version used in `replace` shouldn't matter. We're mostly concerned about the conformance tests.